File: frontend/metropolitan/src/app/shared/model/counter.ts
export interface Counter {
  id?: number;
  name?: string;
  nombreCod?: string;
  telefono: string;
  direccion: string;
  porcentajeNormal: number;
  porcentajeLow: number;
  porcentajeCorp: number;
  porcentajeEspecial: number;
  idAgencia?: string;
  idsProveedores?: string;
  idSucursal: number;
  createBy?: number;
  modify?: number;
  createDate?: string;
  modifyDate?: string;
}

File: backend/Metropolitan/Common/model/counter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataBase.model
{
    public class Counter
    {
        public int id;
        public string name;
        public string nombreCod;
        public string direccion;
        public string telefono;
        public double porcentajeNormal;
        public double porcentajeLow;
        public double porcentajeCorp;
        public double porcentajeEspecial;
        public int idAgencia;
        public string idsProveedores;
        public int idSucursal;
        public int createBy;
        public int modify;
        public DateTime createDate;
        public DateTime modifyDate;
    }

    public class reportByCounter
    {
        public int id;
        public string agencia;
        public string nombre;
        public double total;
        public double totalSales;
    }

    public class reportByCounterDetalle
    {
        public int idNota;
        public string agencia;
        public string nombreCounter;
        public double totalCounter;
        public double totalSales;
        public double? tipoCambioValor;
    }
}
